### DISCUSSION TITLE
TLS demo establish communication fail

### DISCUSSION THREAD

**Discussion by buyobuyo404** [open-quantum-safe/openssl#337] [Q&A] [answered]
I followed the steps to build and run in OQS-OpenSSL_1_1_1-stable. But, when I test TLS demo, the client and server cannot establish communication. Client cannot send any message like strings to server. How can I fix it, or what should I do next step? Thank you for your help!

ps. the following commands are what I ran in my ubuntu 18.04:

1. generated the root CA

   ```linux
   apps/openssl req -x509 -new -newkey dilithium2 -keyout dilithium2_CA.key -out dilithium2_CA.crt -nodes -subj "/CN=oqstest CA" -days 365 -config apps/openssl.cnf
   ```

2. The server generates its key pair

   ```
   apps/openssl genpkey -algorithm dilithium2 -out dilithium2_srv.key
   ```

3. The server generates a certificate request and sends it the to CA

   ```
   apps/openssl req -new -newkey dilithium2 -keyout dilithium2_srv.key -out dilithium2_srv.csr -nodes -subj "/CN=oqstest server" -config apps/openssl.cnf
   ```

4. The CA generates the signed server certificate

   ```
   apps/openssl x509 -req -in dilithium2_srv.csr -out dilithium2_srv.crt -CA dilithium2_CA.crt -CAkey dilithium2_CA.key -CAcreateserial -days 365
   ```

5. run a TLS server

   ```
   apps/openssl s_server -cert dilithium2_srv.crt -key dilithium2_srv.key -www -tls1_3
   ```

6. run a TLS client

   ```
   apps/openssl s_client -groups bikel1 -CAfile dilithium2_CA.crt
   ```

the following are what the result of running the code: when client send a message, server can not receive

![image-20211123124729692](https://user-images.githubusercontent.com/78010512/142973392-24d98424-2ac0-4f54-a77b-1ae82fbab8fa.png)

**Comment by baentsch** [comment, marked as answer]
The data sent by the client is no valid HTTP command. Accordingly the server, started with option `-www`, will not/cannot respond. Sending an HTTP command like "GET /" in turn will trigger the server logic.

**Reply by buyobuyo404** [reply]
Thanks for your help. I try with "GET /", it works. really appreciate your assistance ^-^

**Reply by Jpgig** [reply]
Can you please share screenshot about how did you used "GET/" command at client node to trigger the server logic?
[Client_Server_node](https://user-images.githubusercontent.com/115164852/195674601-778311b5-6b16-4c7b-b191-f47e7bcfe7fe.jpg)

**Reply by baentsch** [reply]
Just below the output "read R block" type 
GET /
and hit ENTER.

**Reply by Jpgig** [reply]
Thanks for your help.
